src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -109,6 +133,41 @@ async function stopMinikube(context: Context): Promise<void> {
     });
 }
 
+async function minikubeStatus(context: Context, notification: boolean): Promise<MinikubeInfo> {
+    if (!await checkPresent(context, notification? CheckPresentMode.Alert : CheckPresentMode.Silent)) {
+        if (notification) {
+            return;
+        }
+        throw new Error('minikube executable could not be found!');
+    }
+
+    try {
+        const result = await context.shell.exec(
+            `${context.binPath} status --format '["{{.MinikubeStatus}}","{{.ClusterStatus}}","{{.KubeconfigStatus}}"]'`);
+        if (result.stderr.length === 0) {
+            const obj = JSON.parse(result.stdout);
+            if (notification) {
+                vscode.window.showInformationMessage(`Minikube is ${obj[0]}`);
+            }
+            return {
+                running: 'Stopped' !== obj[0],
+                cluster: obj[1],
+                kubectl: obj[2],
+            } as MinikubeInfo;
+        }
+        if (notification) {
+            vscode.window.showErrorMessage(`failed to get status: ${result.stderr}`);
+        } else {
+            throw new Error(`failed to get status: ${result.stderr}`);
+        }
+    } catch (err) {
+        if (notification) {
+            vscode.window.showErrorMessage(`minikube status failed: ${err}`);
+        }
+        throw(err);
```

Comment:
`err` is rethrown even if a message has been displayed.  Won't that result in the message being displayed twice if this bubbles back to the top level?

Reply:
I feel like the intermingling of 'UI mode' and 'silent mode' here is causing duplication and potential bugs.  Something to consider is having the function return a `Promise<Errorable<MinikubeInfo>>` instead - then the consumer of the function could check for an error and display it or swallow it according to its posture (and similarly in the success case could display UI or consume the info programmatically).  This simplifies this function and keeps the UI in a UI-oriented caller.  If you want the UI to be reusable then you could still have a `displayMinikubeStatus` function which called this 'silent' one.